kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableFeaturesSuite.scala

```diff
@@ -317,6 +317,50 @@ class DeltaTableFeaturesSuite extends DeltaTableWriteSuiteBase {
     }
   }
 
+  /* ---- Start: type widening tests ---- */
+  test("only typeWidening feature is enabled when metadata supports it") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      createEmptyTable(
+        engine,
+        tablePath = tablePath,
+        schema = testSchema,
+        tableProperties = Map("delta.enableTypeWidening" -> "true"))
+
+      val protocolV0 = getProtocol(engine, tablePath)
+      assert(!protocolV0.supportsFeature(TableFeatures.TYPE_WIDENING_RW_PREVIEW_FEATURE))
+      assert(protocolV0.supportsFeature(TableFeatures.TYPE_WIDENING_RW_FEATURE))
+
+      // try enabling type widening again and expect no change in protocol
+      updateTableMetadata(
```

Comment:
it seems like maybe we should have an update test that doesn't start out with type widening enabled, to verify only one is enabled (I think this is partially covered with other tests above, but it might be simpler to reason about here)?

Reply:
yep, added one.